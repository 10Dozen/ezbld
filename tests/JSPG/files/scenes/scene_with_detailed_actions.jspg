# CurrentScene

Description line.

@ Action 1 | dialog_left pic.jpg | MyTag1
*goto: "NextScene"
*exec: Foo.Bar()

@ Action 2 | scene_right | MyTag2
*goto: "NextScene"

Description of action.

@ Action 3 || MyTag3
*goto: "NextScene"
**condition:
    Bar.Foo()

@ Action 4 | scene_left

It's over!

It's literally over!
