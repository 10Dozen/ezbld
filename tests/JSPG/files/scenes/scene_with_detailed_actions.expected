[0]
cls = JSPGScene
name = CurrentScene | TEXT
desc = "Description line."

[1]
cls = JSPGAction
name = Action 1  | TEXT
scene = CurrentScene | TEXT
type = dialog_left | TEXT
portrait = pic.jpg | TEXT
tag = MyTag1 | TEXT
goto = NextScene | TEXT
exec = ()=>{ Foo.Bar() } | FUNCTION

[2]
cls = JSPGAction
name = Action 2 | TEXT
scene = CurrentScene | TEXT
type = scene_right | TEXT
tag = MyTag2 | TEXT
goto = NextScene | TEXT
desc = "Description of action."

[3]
cls = JSPGAction
name = Action 3 | TEXT
scene = CurrentScene | TEXT
tag = MyTag3 | TEXT
goto = NextScene  | TEXT
condition =
    ()=>{
    \tBar.Foo()
    } | MULTILINE_FUNCTION

[4]
cls = JSPGAction
name = Action 4 | TEXT
scene = CurrentScene | TEXT
type = scene_left | TEXT
desc =
    "It's over!"

    "It's literally over!"
